### ISSUE TITLE
Pending issues with Limit

### ISSUE THREAD

**Issue by Happypig375** [jiggzson/nerdamer#155]
http://nerdamer.com/#comment-3441441124

Don't let that comment be forgotten!

**Comment by Happypig375** [comment]
P.S. Wait for Disquz to load and you will be directed to the comment!

**Comment by jiggzson** [comment]
Moving to #183

**Comment by jiggzson** [comment]
Agreed. I got a little confused. There a way too many duplicate issues and I need to clean them up so I can come up with a plan of action.

**Comment by jiggzson** [comment]
@Happypig375, this was implemented a little while back but I never updated this issue. See https://nerdamer.com/functions/limit.html.

**Comment by Happypig375** [comment]
`limit((1+1/x)^x*x-e*x,x,Infinity)` => `-Infinity`
Expected: `-e/2`

https://www.quora.com/What-is-the-hardest-limit-math-problem (the 2nd answer)

**Comment by Happypig375** [comment]
Missing active label

**Comment by Happypig375** [comment]
A fourth parameter should be added to decide the direction of approach: any negative number for limit from the left, `0` (default) for two-sided limit, any positive number for limit from the right.

**Comment by jiggzson** [comment]
@Happypig375, my initial approach did just that but I found it difficult to implement. I'm hoping to add this as soon as I'm feeling somewhat confident that we caught all if not most of the bugs.

**Comment by Happypig375** [comment]
limit((2sin(x)-sin(2x))/(x-sin(x)),x,0) => -1
According to https://en.wikipedia.org/wiki/L%27Hôpital%27s_rule#Examples, this should be 6

**Comment by Happypig375** [comment]
limit((e^x+e^-x)/(e^x-e^-x),x,1) => 961279542332782/732104881680479, unable to give Wolfram Alpha's lim_(x->1) (e^x + e^(-x))/(e^x - e^(-x)) = (1/e + e)/(-1/e + e)

**Comment by Happypig375** [comment]
limit((x^(1/2)+x^(-1/2))/(x^(1/2)-x^(-1/2)),x,Infinity) => (-limit(sqrt(x),x,Infinity)^(-1)+limit(sqrt(x),x,Infinity))^(-1)*(limit(sqrt(x),x,Infinity)+limit(sqrt(x),x,Infinity)^(-1))
unable to give 1 according to https://en.wikipedia.org/wiki/L%27Hôpital%27s_rule#Complications

**Comment by Happypig375** [comment]
limit((x+sin(x)cos(x))/(x+sin(x)cos(x))/e^(sin(x)), x, Infinity) => 119696244/325368125 but it should not be solvable according to https://en.wikipedia.org/wiki/L%27Hôpital%27s_rule#Counterexamples_when_the_derivative_of_the_denominator_is_zero

**Comment by jiggzson** [comment]
The irony is that I was literally just playing around with a few limits last night and realized that some of them give the incorrect answer. Nerdamer shows some as divergent but they aren't for example `limit(cos(x)/x^2,x,0)`.

**Comment by jiggzson** [comment]
@Happypig375 , `limit((x^(1/2)+x^(-1/2))/(x^(1/2)-x^(-1/2)),x,Infinity) => 1` fixed. Agree?

**Comment by jiggzson** [comment]
> limit((1+1/x)^x*x-e*x,x,Infinity) => -Infinity
Expected: -e/2
>
>https://www.quora.com/What-is-the-hardest-limit-math-problem (the 2nd answer)

@Happypig375, looks like I may have waited a little too long since the steps for this one don't seem to be there anymore. Unless I'm overlooking something.
